### ISSUE TITLE
Is there a reason parameters are Iterators instead of IntoIterator?

### ISSUE THREAD

**Issue by FelixBenning** [BurntSushi/rust-stats#14]
If I understand what I am doing, then, if
```
pub fn mean<I>(it: I) -> f64 where
    I: Iterator,
    <I as Iterator>::Item: ToPrimitive, 
```
was changed to something like
```
fn mean<'a, I, T>(x:I)-> f64 where
    I: IntoIterator<Item= T>,
    T: Into<&'a f64>
{
   let it = x.into_iter();
   ...
}
```

You could also use mean on vectors. e.g.
```
mean(&vec![1,2,3])
```
as opposed to
```
mean(vec![1,2,3].iter().collect())
```

without losing any functionality for iterators themselves. But I am just learning Rust, so I might not understand some limitation of this approach.

**Comment by BurntSushi** [comment]
Yes, the API should use `IntoIterator`. This crate was actually written `IntoIterator` existed (IIRC), and I just never updated the APIs.

But you don't need `vec![1, 2, 3].iter().collect()`. Just `&[1, 2, 3].iter()` should work. Or at least, `&[1, 2, 3].iter().cloned()`.
